flex-backend/api/http/server.go

```diff
@@ -15,50 +15,76 @@ type Server struct {
 }
 
 func (s Server) BuildEndpoints() {
-	http.HandleFunc("/getMovies", handleGetMovies)
-	http.HandleFunc("/getInfo", handleGetMovieInfo)
+	http.HandleFunc("/getMovies", s.handleGetMovies)
+	http.HandleFunc("/getInfo", s.handleGetMovieInfo)
 	http.HandleFunc("/playFile", handlePlayFile)
 	http.HandleFunc("/stopFile", handleStopFile)
 	http.HandleFunc("/ws", s.handleWebSocket)
 	// Used to test the websocket
 	http.HandleFunc("/", handleSendRoot)
 }
 
-func (Server) Serve() {
+func (Server) Serve(port int) {
 	// Start the server
-	log.Println("Starting server on port 8080")
-	log.Fatal(http.ListenAndServe(":8080", nil))
+	log.Printf("Starting server on port %d\n", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", port), nil))
 }
 
 func handleSendRoot(w http.ResponseWriter, r *http.Request) {
-	// list out local movies
-	fmt.Println("Called route!")
 	http.ServeFile(w, r, "websockets.html")
 }
 
-func handleGetMovies(w http.ResponseWriter, r *http.Request) {
-	// list out local movies
-	fmt.Fprintf(w, "Hello, %q", html.EscapeString(r.URL.Path))
-	fmt.Println("Called get movies route!")
+func (s Server) handleGetMovies(w http.ResponseWriter, r *http.Request) {
+	// Send back the filepaths to client as json
+	files, err := s.MovieHandler.ReadLocalDir(s.MovieHandler.MovieDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	jData, err := json.Marshal(files)
+	if err != nil {
+		log.Println(err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(jData)
 }
 
-func handleGetMovieInfo(w http.ResponseWriter, r *http.Request) {
+func (s Server) handleGetMovieInfo(w http.ResponseWriter, r *http.Request) {
 	// Get video file from request, and send back what we know about it.
-	// Ex name of file with no extension, length, cover art, etc.
-	fmt.Fprintf(w, "Hello, %q", html.EscapeString(r.URL.Path))
-	fmt.Println("Called get movie info route!")
-}
+	// Query param will need to have the full file path, since that is what we are sending to the client when it asks for the list of movies
+	var movie string
+	for k, v := range r.URL.Query() {
```

Comment:
We should have all of the request data in the body of the request then unmarshal it to a struct so that we don't need to have conditional logic that is going to put weird branches in our code. The way that the default go router works is it does not support http verbs. so this can just not be a get request (get requests cant have a body)

Reply:
@kcpetersen111  I don't think I am understanding here. The way this currently works is it just looks at the query URL, ex `localhost:8080/getInfo?movie=1&movie=2` so we aren't having a request body since it is just the url that is going to be giving us the info, unless I am wrong here. Do you have an example of what you are trying to explain so I could take a look and maybe gain more insight?